src/app/subscription.ts

```diff
@@ -1,46 +1,42 @@
 import {
   Component,
-  Input,
-  FORM_DIRECTIVES,
-  CORE_DIRECTIVES,
-  AfterViewChecked,
-  OnInit,
-  OnDestroy,
+  Input
 } from 'angular2/angular2';
 
 @Component({
   selector: 'subscription',
   templateUrl: 'src/app/subscription.html',
-  inputs: ['search', 'pusher'],
-  directives: [CORE_DIRECTIVES]
+  inputs: ['search', 'pusher']
 })
-export default class SubscriptionComponent implements AfterViewChecked, OnDestroy, OnInit {
-  @Input() search: string;
+export default class SubscriptionComponent {
+  @Input() search: any;
   @Input() pusher;
   public tweets : Object[];
   private channel;
 
-  public onInit() {
+  ngOnInit() {
     this.subscribeToChannel();
     this.tweets = [];
   }
 
-  private subscribeToChannel() {
-    var encoded = btoa(this.search);
+  subscribeToChannel() {
+    var encoded = btoa(this.search.term);
     this.channel = this.pusher.subscribe(encoded);
-    this.channel.bind('new_tweet', this.newTweet.bind(this));
+    this.channel.bind('new_tweet', (data) => {
+      this.newTweet(data);
+    });
   }
 
-  private newTweet(data: Object) {
+  newTweet(data: Object) {
     this.tweets.push(data);
   }
 
-  public onDestroy() {
+  ngOnDestroy() {
```

Comment:
The interfaces were fine either way I believe, but the methods got renamed in .47:

https://github.com/angular/angular/commit/604c8bbad54ffbcec1d6a5e407821adda8a74a19